src/integrationTest/java/uk/gov/hmcts/reform/pdf/service/integration/GeneratedPDFContentV2Test.java

```diff
@@ -44,30 +64,51 @@ private static String textContentOf(byte[] pdfData) throws IOException {
         }
     }
 
-    @SuppressWarnings("unchecked")
+    @After
+    public void verifyAuth() {
+        verify(authService).authenticate("some-service-auth-header");
+    }
+
     @Test
     public void shouldCreateExpectedPdfFromPlainHtmlTemplate() throws Exception {
-        assertThatGeneratedPdfContains("<html><body>Hello!</body></html>", Collections.EMPTY_MAP, "Hello!");
+        HttpServletResponse response = getResponse("<html><body>Hello!</body></html>", Collections.emptyMap());
+        assertGeneratedPdfContains(response, "Hello!");
     }
 
     @Test
     public void shouldCreateExpectedPdfFromPlainTwigTemplateAndPlaceholders() throws Exception {
-        Map<String, Object> values = ImmutableMap.of("hello", "World!");
-        assertThatGeneratedPdfContains("<html>{{ hello }}</html>", values, "World!");
+        HttpServletResponse response = getResponse("<html>{{ hello }}</html>", ImmutableMap.of("hello", "World!"));
+        assertGeneratedPdfContains(response, "World!");
     }
 
-    private void assertThatGeneratedPdfContains(String template, Map<String, Object> values, String expectedText)
-        throws Exception {
+    @Test
+    public void failAuthentication() throws Exception {
+        Response feignResponse = Response.builder().headers(Collections.emptyMap()).status(666).build();
```

Comment:
It would be best if the name said sth like `shouldFailAuthenticationWhenAuthServiceThrowsException`, which describes the edge case we're testing.

Also, maybe it would be better to change the status. I don't find it controversial or anything, but I imagine there are people out there who might. I don't know, up to you.